File: API/RequestHelpers/ProductParams.cs
namespace API.RequestHelpers
{
    public class ProductParams : PaginationParams
    {
        public string? OrderBy    { get; set; }   // e.g. price | priceDesc | name
        public string? SearchTerm { get; set; }   // Partial match keyword
        public string? Brands     { get; set; }   // Comma-separated list, e.g. "Angular,React"
        public string? Types      { get; set; }   // Comma-separated list, e.g. "Boots,Hats"
    }
}

File: client/src/app/models/productParams.ts
// src/app/models/productParams.ts
export type ProductParams = {
    orderBy: string;           // e.g. "name", "price", "priceDesc"
    searchTerm: string;
    types: string[];
    brands: string[];
    pageNumber: number;
    pageSize: number;
  };
